test/test_custom_function.py

```diff
@@ -43,8 +43,8 @@ def forward(self, a:LazyBuffer, b:LazyBuffer) -> LazyBuffer:
                              arg={"GPU": atan2_gpu, "CPU": atan2_cpu}[a.device], srcs=(a.contiguous(), b.contiguous()))
   def backward(self, grad_output:LazyBuffer) -> Tuple[Optional[LazyBuffer], Optional[LazyBuffer]]:
     denom = (self.a.e(BinaryOps.MUL, self.a)).e(BinaryOps.ADD, self.b.e(BinaryOps.MUL, self.b))
-    return grad_output.e(BinaryOps.MUL, self.b.e(BinaryOps.DIV, denom)) if self.needs_input_grad[0] else None, \
-           grad_output.e(BinaryOps.MUL, self.a.const(0).e(BinaryOps.SUB, self.a).e(BinaryOps.DIV, denom)) if self.needs_input_grad[1] else None
+    return grad_output.e(BinaryOps.MUL, self.b.e(BinaryOps.MUL, denom.e(UnaryOps.RECIP))) if self.needs_input_grad[0] else None, \
+           grad_output.e(BinaryOps.MUL, self.a.const(0).e(BinaryOps.SUB, self.a).e(BinaryOps.MUL, denom.e(UnaryOps.RECIP))) if self.needs_input_grad[1] else None # noqa: E501
```

Comment:
missed this earlier, we don't want new `# noqa: E501`

Reply:
I have this modification on pattern matcher.

`(UPat(UOps.ALU, BinaryOps.MUL, [UPat(name="x"), UPat(UOps.CONST, 0, name="c")]), lambda x,c: c if not isinstance(x.arg, (int, float)) else x if math.isnan(x.arg) else c)`

I need that two how the x*0 unfolds work, I try to create a new rule the specific x=nan but don't work. 🤔

Reply:
I try this rule

```
(UPat(UOps.ALU, BinaryOps.MUL, [UPat(UOps.CONST, math. nan, name="x"), UPat()]), lambda x: x),
```

Without modifying the original, and throws errors, Can I have two rules for the same Ops but different `CONST`? 🤔 

<img width="848" alt="Screenshot 2024-06-11 at 7 10 31 PM" src="https://github.com/tinygrad/tinygrad/assets/32320832/53fae38c-617a-4d4f-86d3-6178125ce0b5">

Reply:
Ignored, just rewrite the whole line. Testing now

Reply:
I change the `math` import to just import isnan function, is the unique use here. and pass the check. Is that valid? Also I rewrite the whole line to

`(UPat(UOps.ALU, BinaryOps.MUL, [UPat(name="x"), UPat(UOps.CONST, 0, name="c")]), lambda x,c: x if isinstance(x.arg, float) and isnan(x.arg) else c),`

Reply:
you don't need to change the algorithm, the point is that we don't a very long line as it's less readable

Reply:
If I leave the math import merged with the next one, the line need a E501, so in this case is valid to left a E501 in this line, if understand correctly?

Reply:
no, just break the line somehow, either parentheses or `\`

Reply:
Ohh got the point, thx for the explanation.